internal/build/phase_config_provider.go

```diff
@@ -108,11 +110,25 @@ func WithBinds(binds ...string) PhaseConfigProviderOperation {
 func WithDaemonAccess() PhaseConfigProviderOperation {
 	return func(provider *PhaseConfigProvider) {
 		WithRoot()(provider)
+
 		bind := "/var/run/docker.sock:/var/run/docker.sock"
+
+		if dockerHost, ok := os.LookupEnv("DOCKER_HOST"); ok {
+			if strings.HasPrefix(dockerHost, "unix://") {
+				bind = fmt.Sprintf("%s:/var/run/docker.sock", strings.TrimPrefix(dockerHost, "unix://"))
+			}
+			if strings.HasPrefix(dockerHost, "tcp://") {
+				bind = ""
+				provider.hostConf.NetworkMode = "host"
```

Comment:
I am not sure how well this would interact with `WithNetwork()`.

Reply:
Maybe I should check if the NetworkMode is non-empty string. If it's empty then it's ok, if not I should print warning?

Reply:
Also this probably will only work with Linux.